packages/flutter/lib/src/material/platform_builder.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Builds different widgets according to the current platform environment.
+///
+/// When specified, [themeAdaptiveType] can be used to selectively adapt
+/// to the platform only if an ancestor [Theme] specified the widget type
+/// in [ThemeData.adaptiveWidgetTheme].
+///
+/// See also:
+///
+///  * [AdaptiveWidgetThemeData.bundled], for a list of Material Design widgets
+///    that can adapt using Cupertino widgets on iOS.
+class PlatformBuilder extends StatelessWidget {
```

Comment:
The way this is defined doesn't seem to jibe with the name.  Shouldn't it just have a single build parameter, a function value that's passed the BuildContext and TargetPlatform?